### ISSUE TITLE
[BUG] Canceling GitHub login flow results in generic 404 error page.

### ISSUE THREAD

**Issue by Ferroin** [netdata/netdata-cloud#36]
**Describe the bug**	

When attempting to log in to Netdata Cloud using a GitHub Account when you have not already linked a GitHub Account to your Netdata Cloud account, canceling the authentication flow from the GitHub side results in a generic 404 error page in Netdata Cloud.

**To Reproduce**	

1. Ensure you do not have a GitHub account linked to your Netdata Cloud account (or do not have a Netdata Cloud account) and are logged out of Netdata Cloud.
2. Go to https://app.netdata.cloud/
3. Click the ‘Continue with GitHub’ button.
4. When the GitHub OAuth page comes up, click ‘Cancel’ instead of ‘Authorize’.

**Expected behavior**	

The user should be presented with a dedicated error page for this situation with a link to return to the login page.

**Screenshots**	

n/a

**Error logs**	

https://aurek.ahferroin7.net/pub/app.netdata.cloud.har

**Desktop:**

 - OS: Windows 10 20H2
 - Browser: Google Chrome	
 - Version 87.0.4280.141

Also confirmed on Microsoft Edge 87.0.664.75 on the same system.

**Comment by allelos** [comment]
Guys, after seeing the url we are returned to after clicking cancel [here](https://app.netdata.cloud/api?error=access_denied&error_description=The+user+has+denied+your+application+access.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23access-denied&state=6omslw0TUBiJOpfxZ7w0A) and according to the github oauth documentation regarding application suspension [here](https://docs.github.com/en/developers/apps/troubleshooting-authorization-request-errors) I am assuming that the callback url isn't configured correctly on the BE side. @lvrach can you validate if that's the case?

**Comment by dimko** [comment]
@chrikar Could you please check if this is an easy to fix thing with no great effort?

**Comment by dimko** [comment]
Assigning @dionysis-netdata to re-evaluate in the future what needs to be done

**Comment by cakrit** [comment]
@vkuznecovas / @ravilushqa this should be trivial to handle, seems like no one bothered to look into it.

**Comment by ravilushqa** [comment]
thanks @cakrit and @Ferroin

I put the task in the backlog

**Comment by ravilushqa** [comment]
after clicking cancel we are redirecting to callback, this parameter is not passing from the backend, it was set up at creating oauth app page. https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app
Can someone who has permission prove that callback for the app is: https://app.netdata.cloud/api 

@cakrit @amalkov

**Comment by cakrit** [comment]
I saw that the callback URL is https://netdata.cloud/api/v1/auth/github/callback

Contacted @ravilushqa and he said not to change it, but has questions I can't answer:

Am I understanding correctly that our authorization app is related to https://github.com/netdatabot?
Do we use this OAuth application only in one place or do we have other flows that could be affected by editing this app?

It says that netdata owns this app and that there are two users for it, so probably two separate applications with their own tokens each, or sharing the same token, not sure how it works.

**Comment by ravilushqa** [comment]
Here is redirect url to github
```
https://github.com/login/oauth/authorize?client_id=56abb21b06399422e13d&redirect_uri=https://app.netdata.cloud/api/v2/auth/account/github/callback&response_type=code&scope=read:user user:email&state={state}
```
Intresting part is redirect_uri. Github documentation [says](https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#redirect-urls):

> The redirect URL's path must reference a subdirectory of the callback URL.

If we compare callback URL that @cakrit mention above and our url from real request, we can see that they are not right. I have a feeling that provided callback URL from @cakrit is not actual that is used. So probably we have another github application for it. Also I've found that actual client id for application is `56abb21b06399422e13d`. Can we prove that this client is related to mentioned app above or find related one?

**Comment by cakrit** [comment]
The client id in the Netdata org is `d7bde496f84f1f801cf3`.
Perhaps someone thought that `netdatabot` as a user should own the OAuth application and created it there? I can't login as `netdatabot` though, so I can't verify this hypothesis.

**Comment by ravilushqa** [comment]
@novykh yes, you right. However redirect_uri doesn't work for the error flow. We are sending valid redirec_uri, but we are redirecting just to the callback url, and as we don't handle it we get 404
